From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Tue, 15 Aug 2023 00:13:34 +0200
Subject: quickSettings: Add getFirstItem() method

We will need some more access to the menu's underlying grid to
provide extension API for adding additional quick items.

Expose a new getFirstItem() method that (surprise!) returns the
first item.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2894>
(cherry picked from commit 96bf9e700f79457b73381f8d367d516723e2cb47)
---
 js/ui/quickSettings.js | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/js/ui/quickSettings.js b/js/ui/quickSettings.js
index c684141..1bc84a0 100644
--- a/js/ui/quickSettings.js
+++ b/js/ui/quickSettings.js
@@ -802,6 +802,10 @@ export const QuickSettingsMenu = class extends PopupMenu.PopupMenu {
         }
     }
 
+    getFirstItem() {
+        return this._grid.get_first_child();
+    }
+
     open(animate) {
         this.actor.show();
         super.open(animate);
